pkg/watch/until.go

```diff
@@ -23,7 +23,9 @@ import (
 )
 
 // ConditionFunc returns true if the condition has been reached, false if it has not been reached yet,
-// or an error if the condition cannot be checked and should terminate.
+// or an error if the condition cannot be checked and should terminate. In general, it is better to define
+// level driven conditions over edge driven conditions (pod has ready=true, vs pod modified and ready changed
+// from false to true).
```

Comment:
You can see a bit how I'm using this in #23567 to check one condition
then move to the other.  I have the invariants check to look at the
whole watch stream and look for regression.

But it's worth noting that we do have edge conditions to test for -
for instance, readiness can be reset if liveness fires.  If I want to
string together conditions to test that I'll end up using some edges.
Even our controllers have *some* edges (that are rechecked later).